hostfactory/1.1/providerplugins/azurecc/scripts/src/cyclecloud_provider.py

```diff
@@ -1088,16 +1129,25 @@ def _terminate_expired_requests(self):
             for request_id, request in requests_store.items():
                 if request.get("completed"):
                     continue
-
-                if request.get("allNodes") is None:
-                    logger.warning("Yet to find any NodeIds for RequestId %s", request_id)
-                    continue
-
+                
                 created_timestamp = request["requestTime"]
                 now = calendar.timegm(self.clock())
                 delta = now - created_timestamp
 
                 if delta > self.creation_request_ttl:
+                    if request.get("allNodes") is None:
+                        logger.warning("Yet to find any NodeIds for RequestId %s", request_id)
+                        try:
+                            nodes_by_request_id = {}
+                            nodes_by_request_id.update(self.cluster.nodes(request_ids=[request_id]))   
+                            logger.debug("Node list by request id %s",nodes_by_request_id)             
```

Comment:
Why aren't we updating request["allNodes"]?

Reply:
yes this should be updated here. But now we are not seeing this error so often "Yet to find any NodeIds for RequestId" but its a bug